lib/pka_queue.c

```diff
@@ -891,9 +930,11 @@ int pka_queue_cmd_dequeue(pka_queue_t            *queue,
     return 0;
 }
 
-int pka_queue_rslt_dequeue(pka_queue_t            *queue,
-                           pka_queue_rslt_desc_t  *rslt_desc,
-                           pka_results_t          *results)
+
+int pka_queue_rslt_dequeue_by_user_data(pka_queue_t            *queue,
+                                        pka_queue_rslt_desc_t  *rslt_desc,
+                                        pka_results_t          *results,
+                                        void                   *user_data)

```

Comment:
This user_data argument is not used in this implementation?

Reply:
It’s reserved to check for results came back out of sequence. With new locking mechanism and testing in both synchronous/asynchronous mode, out of sequence results are not seen.